hack/make-rules/test-cmd-util.sh

```diff
@@ -4188,7 +4188,7 @@ run_plugins_tests() {
   kube::test::if_has_string "${output_message}" 'no plugins installed'
 
   # single plugins path
-  output_message=$(KUBECTL_PLUGINS_PATH=test/fixtures/pkg/kubectl/plugins kubectl plugin 2>&1)
+  output_message=$(! KUBECTL_PLUGINS_PATH=test/fixtures/pkg/kubectl/plugins kubectl plugin 2>&1)
```

Comment:
I'm not feeling good about this change. Why not checking `$?`?

Reply:
What are you worried about @mengqiy ?

Reply:
`record_command` will check every command's `$?`, so for command with no arguments provided, we need to revert the value of `$?` so that `record_command` can be pass,

Reply:
maybe not every command, but it will check the last value of $? in each functions, iirc

Reply:
> What are you worried about @mengqiy ?

Maybe this is fine.
I'm not sure what is the best practice here to check the exit code and output at the same time.